File: 0242. Valid Anagram/main.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
let isAnagram = function (s, t) {
  let len_s = s.length,
    len_t = t.length;
  if (len_s !== len_t) return false;
  const arr_s = new Array(26).fill(0);
  const arr_t = new Array(26).fill(0);
  for (let i = 0; i < len_s; i++) {
    arr_s[s[i].charCodeAt() - 97]++;
    arr_t[t[i].charCodeAt() - 97]++;
  }
  for (let i = 0; i < 26; i++) {
    if (arr_s[i] !== arr_t[i]) return false;
  }
  return true;
};

File: 0242. Valid Anagram/main.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

/**
 * I: s = "anagram", t = "nagaram"
 * O: 1
 *
 * I: s = "rat", t = "car"
 * O: 0
 */

bool isAnagram(string s, string t);

int main(int argc, char const *argv[]) {
    return 0;
}

bool isAnagram(string s, string t) {
    unordered_map<char, int> mp_s;
    unordered_map<char, int> mp_t;
    int len_s = s.size(), len_t = t.size();
    if (len_s != len_t) return false;
    for (int i = 0; i < len_s; i++) {
        mp_s[s[i]]++;
        mp_t[t[i]]++;
    }
    for (auto it : mp_t) {
        if (mp_s.find(it.first) == mp_s.end()) return false;
        if (it.second != mp_s[it.first]) return false;
    }
    return true;
}
